src/Microsoft.OData.Core/ODataMessageReader.cs

```diff
@@ -904,24 +904,18 @@ private ODataMessageInfo GetOrCreateMessageInfo(Stream messageStream, bool isAsy
         {
             if (this.messageInfo == null)
             {
-                if (this.serviceProvider == null)

```

Comment:
so, why do we still keep "https://github.com/OData/odata.net/blob/main/src/Microsoft.OData.Core/ODataServiceCollectionExtensions.cs#L52" ?

Reply:
Strictly speaking, that could be a breaking change. The likelihood of breaking existing code is low, likely 0, but here's sample code that will break today if we removed `ODataMessageInfo` from the DI:

```c#
var serviceCollection = new ServiceCollection();
serviceCollection.AddDefaultODataServices();
var serviceProvider = serviceCollection.BuildServiceProvider();
var messageInfo = serviceProvider.GetRequiredService<ODataMessageInfo>();
```
In the current version, this would return a messageInfo instance, if we remove it from DI and ship a new version, then this code would suddenly throw an exception when the user updates.

If we want to be strict about non-breaking minor and patch releases, we can remove it from the next major version and consider making the class private.

Reply:
Resolved

Reply:
It could be a breaking change but it seems it could have very few users since a search shows that this was only being used in that location. Also the DI version is scoped which should not reproduce this bug/issue.

Reply:
https://github.com/OData/odata.net/blob/main/buildandtest.yml#L24 this build step is the one that fails to build